Prevent linking against installed libraries by telling the linker to look into
LIBRARY_OUTPUT_PATH first. Linker options (-k) are passed to the linker before
a linker script with predefined SEARCH_DIR order.

--- a/hedgewars/CMakeLists.txt
+++ b/hedgewars/CMakeLists.txt
@@ -125,7 +125,7 @@ elseif(${CMAKE_Pascal_COMPILER_VERSION} VERSION_GREATER 2.4)
 endif()
 
 #generic folder where our libraries reside
-add_flag_append(CMAKE_Pascal_FLAGS "-Fl${LIBRARY_OUTPUT_PATH}")
+add_flag_append(CMAKE_Pascal_FLAGS "-k-L${LIBRARY_OUTPUT_PATH}")
 
 #DEPENDECIES AND EXECUTABLES SECTION
 if(APPLE AND (NOT BUILD_ENGINE_LIBRARY))
@@ -154,7 +154,7 @@ if(NOT WIN32)
         list(INSERT engine_sources 0 PNGh.pas)
         list(REMOVE_AT PNG_LIBRARIES 1) #removing the zlib library path
         get_filename_component(PNG_LIBRARY_DIR ${PNG_LIBRARIES} PATH)
-        add_flag_append(CMAKE_Pascal_FLAGS "-k-L${PNG_LIBRARY_DIR} -Fl${PNG_LIBRARY_DIR}")
+        add_flag_append(CMAKE_Pascal_FLAGS "-Fl${PNG_LIBRARY_DIR}")
         add_definitions(-dPNG_SCREENSHOTS)
     endif()
 endif()
